### PR TITLE
Awood/retry collect

### PR DESCRIPTION
**Pull request by awood**

### CODE REVIEW

File: src/main/java/org/candlepin/subscriptions/ApplicationConfiguration.java [hunk 1]
```diff
@@ -141,4 +146,19 @@ public void addResourceHandlers(ResourceHandlerRegistry registry) {
...
     }
 
+
+    @Bean(name = "collectorRetryTemplate")
+    public RetryTemplate retryTemplate() {
+        SimpleRetryPolicy retryPolicy = new SimpleRetryPolicy();
+        retryPolicy.setMaxAttempts(4);
+
+        FixedBackOffPolicy backOffPolicy = new FixedBackOffPolicy();
+        backOffPolicy.setBackOffPeriod(2000L);
```
Comment (mstead) [inline src/main/java/org/candlepin/subscriptions/ApplicationConfiguration.java]: Do you think it is worth making this configurable? I'm undecided.

**Comment by mstead** [comment]
Overall, this seems fine to me and looks like it does what we talked about.

File: src/main/java/org/candlepin/subscriptions/ApplicationConfiguration.java (hunk 1 above)
Comment (kahowell) [reply src/main/java/org/candlepin/subscriptions/ApplicationConfiguration.java]: Meh, for now. Can revisit later.

Comment (mstead) [reply src/main/java/org/candlepin/subscriptions/ApplicationConfiguration.java]: Ya, that's the camp I'm in as well. Figured I'd ask for other's opinions.

**Comment by mstead** [comment]
> I dig this a lot as if some accounts fail to collect, we'll log it and move on, so then we have partial success and can investigate/work on those that fail.

:+1:

**Pull request merged**